mongoengine/queryset/visitor.py

```diff
@@ -101,13 +103,15 @@ def _combine(self, other, operation):
             return self
 
         # Or if this Q is empty, ignore it and just use `other`.
-        if self.empty:
+        if not self:
             return other
 
         return QCombination(operation, [self, other])
 
     @property
     def empty(self):
+        msg = "'empty' property is deprecated in favour of using 'not bool(filter)"
```

Comment:
I actually wonder what would be the best way of handling this field.
1. We can leave line 104 as it was, but deprecation warning will be logged each time fires
2. We can raise `NotImplementedError` here, but that would be a breaking change

This is the case for someone, who was using QNode directly in his project.
Is there a preferred way of handling that?

Reply:
I think its best to leave it was

Reply:
Right. Do we want to log a deprecation warning?

Reply:
yes please

Reply:
@erdenezul Fixed

Reply:
Could you document the deprecation in the changelog?

Reply:
@bagerard Changelog updated, added an example of the transition.